### ISSUE TITLE
Minor bug when order item list contains products which is not "connected" to any one product within a store.

### ISSUE THREAD

**Issue by henjak** [krokedil/klarna-order-management-for-woocommerce#112]
This code generates a critical error in some cases, because it tries to use $product->get_permalink() on bool.

https://github.com/krokedil/klarna-order-management-for-woocommerce/blob/master/includes/klarna-order-management-functions.php#L43-L66

**Fix:** 

```
/**
 * Get the product and its image URLs.
 *
 * @param WC_Order_Item_Product $item The order item.
 * @return array The product and image URL if available, otherwise an empty array.
 */
function kom_maybe_add_product_urls( $item ) {
	$product_data = array();
	$settings     = get_option( 'woocommerce_kco_settings', array() );
	if ( isset( $settings['send_product_urls'] ) && 'yes' === $settings['send_product_urls'] ) {
		$product = wc_get_product( $item->get_product_id() );

		if ( $product instanceof WC_Product ) {
			if ( $product->get_image_id() > 0 ) {
				$image_id                  = $product->get_image_id();
				$image_url                 = wp_get_attachment_image_url( $image_id, 'shop_single', false );
				$product_data['image_url'] = $image_url;
			}

			$product_data['product_url'] = $product->get_permalink();
		}
	}
	return $product_data;
}
```

I moved **$product_data['product_url'] = $product->get_permalink();** inside the if-check, to make sure the $product is an instance of WC_Product.

I've noticed this when I importered two orders into a store, which had no connected products, and tried to sync it to Klarna.

Woo: 8.5.2
WP: 6.4.3
PHP: 8.1

**Comment by mntzrr** [comment]
Hello @henjak ! Thanks for reporting this issue. I expect this issue to have been resolved with the release of #104 .

**Comment by henjak** [comment]
Hello @mntzrr ! Thank you for the reply and fix.
